src/main/java/htsjdk/samtools/BAMRecord.java

```diff
@@ -256,6 +283,55 @@ public Cigar getCigar() {
         return super.getCigar();
     }
 
+    private static boolean isSentinelCigar(final Cigar cigar, final int readLength) {
```

Comment:
are SEQ allowed to be '*' when the cigar is populated? if so, this will not work in that case as it looks at the read-length for guidance. If folks think that this is too restrictive, I can just look for S and N operators and remove the readlength check.

Reply:
Perhaps change the condition to:
```java
return cigar.getCigarElement(0).getOperator() == CigarOperator.S &&
       (cigar.getCigarElement(0).getLength() == readLength || readLength == 0) &&
       cigar.getCigarElement(1).getOperator() == CigarOperator.N;
```

Reply:
ah. I ignored this caveat. I thought that the sentinel is always xSyN

Reply:
Actually I am not familiar with this part. If SEQ='*', is readLength equal to 0? If so, your old initial code is ok.

Reply:
From @yfarjoun:

> Got it. Did we say that in the spec pr?

I don't know. This limitation is implicit in BAM (like chromosome lengths can't be longer than 2Gbp), but I agree it would be good to make it explicit anyway.